tests/test_askrene.py

```diff
@@ -109,6 +107,29 @@ def test_layers(node_factory):
     assert listlayers == {'layers': [expect]}
 
 
+def check_route_as_expected(routes, paths):
```

Comment:
I am not sure what is being tested here. A single hop payment does not pay fees, why does it matter if `auto.sourcefree` is used or not?